src/workspaces.c

```diff
@@ -388,15 +388,79 @@ workspaces_find(struct workspace *anchor, const char *name, bool wrap)
 	return NULL;
 }
 
+static void
+destroy_workspace(struct workspace *workspace)
+{
+	wlr_scene_node_destroy(&workspace->tree->node);
+	zfree(workspace->name);
+	wl_list_remove(&workspace->link);
+	free(workspace);
+}
+
+void
+workspaces_reconfigure(struct server *server)
+{
+	/*
+	 * Currently, we consider updates like below as "WS2 is renamed to WS3
+	 * and WS2 is added":
+	 * old workspaces = ["WS1", "WS2"]
+	 * new workspaces = ["WS1", "WS3", "WS2"]
+	 */
+	struct workspace *conf;
+	struct wl_list *ws_link = server->workspaces.next;
+	wl_list_for_each(conf, &rc.workspace_config.workspaces, link) {
+		if (ws_link == &server->workspaces) {
+			/* # of workspaces increased */
```

Comment:
Suggest adding the work "configured" before workspaces here.

Reply:
Done.